backend/cmm_helpers.ml

```diff
@@ -2526,172 +2566,202 @@ let tuplify_function arity =
 
 let max_arity_optimized = 15
 
-let final_curry_function ~nlocal ~arity =
+let machtype_stored_size t =
+  Array.fold_left
+    (fun cur c ->
+      match c with
+      | Addr -> Misc.fatal_error "[Addr] cannot be stored"
+      | Val | Int -> cur + 1
+      | Float -> cur + if Arch.size_int = 4 then 2 else 1)
+    0 t
+
+let machtype_non_scanned_size t =
+  Array.fold_left
+    (fun cur c ->
+      match c with
+      | Addr -> Misc.fatal_error "[Addr] cannot be stored"
+      | Val -> cur
+      | Int -> cur + 1
+      | Float -> cur + if Arch.size_int = 4 then 2 else 1)
+    0 t
+
+let layout_machtype_for_closure t v =
```

Comment:
Let's find a better name for this function.

Reply:
What about `allocate_value_in_closure`?

Reply:
Indeed, I wondered about the impending terminology problem here when looking at the "any value" layout.  Maybe `value_slot_given_machtype`?

Reply:
(yes I know "value slot" involves "value" but at least it then matches up with existing terminology, which we could rename everywhere if necessary!)